packages/@ionic/cli/src/commands/login.ts

```diff
@@ -170,12 +170,15 @@ If you are having issues logging in, please get in touch with our Support[^suppo
       this.env.log.info(`During this process, a browser window will open to authenticate you. Please leave this process running until authentication is complete.`);
       this.env.log.nl();
 
-      const login = await this.env.prompt({
-        type: 'confirm',
-        name: 'continue',
-        message: 'Open the browser to log in to your Ionic account?',
-        default: true,
-      });
+      let login = true;
+      if (!this.env.flags.confirm) {
+        login = await this.env.prompt({
+          type: 'confirm',
+          name: 'continue',
+          message: 'Open the browser to log in to your Ionic account?',
+          default: true,
+        });
+      }
```

Comment:
Is this change actually needed? I believe the prompt will use the default value in non-interactive mode.

Reply:
Looks like it is not. I've removed the unnecessary check